felt/felt.go

```diff
@@ -1,4 +1,4 @@
-package types
+package felt
```

Comment:
IMO this should stay in `types`, to avoid stuttering

Reply:
The problem here is that if you want several implementation of felts, you need to separate the types from felt. Once done, we should be able to rely on an interface.